engine-nextgen/src/main/kotlin/graphql/nadel/enginekt/transform/result/json/JsonNodeExtractor.kt

```diff
@@ -28,6 +30,34 @@ object JsonNodeExtractor {
         }
     }
 
+    /**
+     * Extract the node at the given json node path.
+     */
+    fun getNodeAt(data: JsonMap, path: JsonNodePath): JsonNode? {
+        val rootNode = JsonNode(JsonNodePath.root, data)
+        return getNodeAt(rootNode, path)
+    }
+
+    /**
+     * Extract the node at the given json node path.
+     */
+    fun getNodeAt(rootNode: JsonNode, path: JsonNodePath): JsonNode? {
+        return path.segments.fold(rootNode as JsonNode?) { currentNode, segment ->
+            when (currentNode?.value) {
+                is Map<*, *> -> currentNode.value[segment.value]?.let {
```

Comment:
Can we change these to `AnyMap` which is a type alias for `Map<*, *>`. I've been using them elsewhere in the codebase, open to discussion if you think it's a bad pattern though. Personally I just think it looks nicer 🤷‍♂️.

Reply:
I initially didn't see the point in using the aliases, but I'm starting to get used to them. Happy to make the change